src/YoTennis/Models/GameHandler.cs

```diff
@@ -155,7 +155,23 @@ public void AddEvent(GameEvent gameEvent)
 
             Events.Add(gameEvent);
         }
-        
+
+        private void On(PastMatchEvent gameEvent)
+        {
+
+            if (CurrentState.State != MatchState.NotStarted)
+                throw new InvalidOperationException("Not Expected");
+                        
+            CurrentState.FirstPlayer = gameEvent.FirstPlayer;
+            CurrentState.SecondPlayer = gameEvent.SecondPlayer;
+            CurrentState.MatchStartedAt = gameEvent.Date;
+
+            CurrentState.FirstPlayerUserId = gameEvent.FirstPlayerUserId;
+            CurrentState.SecondPlayerUserId = gameEvent.SecondPlayerUserId;
+
+            CurrentState.Sets = gameEvent.MatchScore;
+
```

Comment:
the handler does not change state, so it will still be `NotStarted`

Reply:
Done

Reply:
It's will hard!